### ISSUE TITLE
Proposal: Adding cartesian to polar conversion functions

### ISSUE THREAD

**Issue by ivanskodje** [godotengine/godot#7263]
*Bugsquad edit:* Repurposing this issue based on https://github.com/godotengine/godot/pull/7285#issuecomment-272994312

@tagcup wrote:
> Conversion between degrees and a general 2D vector doesn't make any sense to me, in mathematical terms: the norm of a 2D vector isn't necessarily equal to 1. You can't convert between two degrees of freedom (Vector2) and one degree of freedom (an angle) in a meaningful way.
> 
> I would rather propose adding functions that converts between polar and cartesian coordinates in 2D: (x,y) <-> (r,θ) where θ is in radians. The API could be something like cartesian_to_polar(x,y) and polar_to_cartesian(r,θ), each returning a vector.
> 
> Or alternatively, simply a polar constructor for Vector2.

--------------

*Original proposal:*

I find that a lot of users (me included) have the need to convert angles (in radian or degrees) into a Vector2 direction.

The current way to get a Vector2 direction from an angle;
```
  var radian_value = PI # 180 degrees
  var my_vec2 = **Vector2(cos(radian_value), sin(radian_value))**
```

I propose that we add new functions such as **rad2vec**(rad_value) and **deg2vec**(deg_value) with the other functions found at the [core](http://docs.godotengine.org/en/stable/classes/class_@gdscript.html).

This will make the process much cleaner for everyone, and we wont have to know/remember this formula. :)

**Comment by bojidar-bg** [comment]
Note that usually using `Vector2(1,0).rotated(rad)` is pretty clean, as long as you get the x and y of the "up" vector correct...

**Comment by viibr** [comment]
Isn't it Vector2(0,1).rotated() ?

**Comment by Zylann** [comment]
Not as clear and efficient as rad2vec (which I ended up writing myself as a helper too).

**Comment by bojidar-bg** [comment]
@viibr Or maybe it isn't? IDK I always try a few times until it fits the expected response... :laughing: 
Anyway, I agree that this would be a nice helper to have in `@GDScript`.

Actually, that's a pretty-much "junior job", as it is pretty simple to implement (though it should be done in vscript as well). The main changes for GDScript would be in `gd_functions.cpp/.h`, and for VScript, they would be in all places and switches the `TEXT_CHAR` enum (for example) is found.

**Comment by jxfaith** [comment]
I'd like to try my hand at this one. New to the project (and github in general), but it's in line with a hobby I've adopted recently (game design). Seems like as good a place as any to start!

**Comment by bojidar-bg** [comment]
@jxfaith Good luck -- even if you forget to do some part of it, we would discuss it in the PR :wink:

**Comment by vnen** [comment]
Solved some time ago by 054a2ac579c4893f46b680b1cb0d2460aa3d9140.
